app/scenes/select_server/select_server.js

```diff
@@ -90,7 +113,7 @@ export default class SelectServer extends Component {
                                 defaultMessage='Proceed'
                             />
                         </Button>
-                        <ErrorText error={this.props.server.error}/>
+                        <ErrorText error={this.props.server.error || this.state.error}/>
```

Comment:
There's a small issue here. We're not resetting the server error when we attempt to select another server url.
The scenario:
Someone enters a plausible server url which we fail to ping. We display the server error. They edit the url, make a goof, and hit Proceed when the url has an invalid format. We'll still display the server url and the user might not realize their mistake.
As I said, a small issue. It's up to you if you want to address that (in this PR).